From ce9c200bd677386deb8b3a254f80039e14e2e666 Mon Sep 17 00:00:00 2001
From: yminer <yminer@vmware.com>
Date: Wed, 5 Jun 2024 08:37:00 +0000
Subject: [PATCH] test beego v2.2.1

Signed-off-by: yminer <yminer@vmware.com>

update artifact dao count

update task dao count

update execution dao count

refactor disableSort option

using Option struct instead of const

remove comments

use option func

simplify

diff --git a/src/lib/orm/query.go b/src/lib/orm/query.go
index 0e193886a93..b30bf47a19f 100644
--- a/src/lib/orm/query.go
+++ b/src/lib/orm/query.go
@@ -64,7 +64,14 @@ import (
 //			},
 //		 }
 //	}
-func QuerySetter(ctx context.Context, model interface{}, query *q.Query) (orm.QuerySeter, error) {
+
+type Config struct {
+	DisableSort bool
+}
+
+type Option func(*Config)
+
+func QuerySetter(ctx context.Context, model interface{}, query *q.Query, options ...Option) (orm.QuerySeter, error) {
 	t := reflect.TypeOf(model)
 	if t.Kind() != reflect.Ptr {
 		return nil, fmt.Errorf("<orm.QuerySetter> cannot use non-ptr model struct `%s`", getFullName(t.Elem()))
@@ -82,8 +89,12 @@ func QuerySetter(ctx context.Context, model interface{}, query *q.Query) (orm.Qu
 	// set filters
 	qs = setFilters(ctx, qs, query, metadata)
 
+	opts := newConfig(options...)
 	// sorting
-	qs = setSorts(qs, query, metadata)
+	// Do not set sort when sort disabled. e.g. for Count()
+	if !opts.DisableSort {
+		qs = setSorts(qs, query, metadata)
+	}
 
 	// pagination
 	if query.PageSize > 0 {
@@ -96,6 +107,20 @@ func QuerySetter(ctx context.Context, model interface{}, query *q.Query) (orm.Qu
 	return qs, nil
 }
 
+func WithSortDisabled(disableSort bool) Option {
+	return func(cfg *Config) {
+		cfg.DisableSort = disableSort
+	}
+}
+
+func newConfig(opts ...Option) *Config {
+	config := &Config{}
+	for _, opt := range opts {
+		opt(config)
+	}
+	return config
+}
+
 // PaginationOnRawSQL append page information to the raw sql
 // It should be called after the order by
 // e.g.
@@ -121,7 +146,7 @@ func QuerySetterForCount(ctx context.Context, model interface{}, query *q.Query,
 	query.Sorts = nil
 	query.PageSize = 0
 	query.PageNumber = 0
-	return QuerySetter(ctx, model, query)
+	return QuerySetter(ctx, model, query, WithSortDisabled(true))
 }
 
 // set filters according to the query
diff --git a/src/pkg/artifact/dao/dao.go b/src/pkg/artifact/dao/dao.go
index 880667e7fa3..0e2e79a41cc 100644
--- a/src/pkg/artifact/dao/dao.go
+++ b/src/pkg/artifact/dao/dao.go
@@ -93,7 +93,7 @@ func (d *dao) Count(ctx context.Context, query *q.Query) (int64, error) {
 			Keywords: query.Keywords,
 		}
 	}
-	qs, err := querySetter(ctx, query)
+	qs, err := querySetter(ctx, query, orm.WithSortDisabled(true))
 	if err != nil {
 		return 0, err
 	}
@@ -282,8 +282,8 @@ func (d *dao) DeleteReferences(ctx context.Context, parentID int64) error {
 	return err
 }
 
-func querySetter(ctx context.Context, query *q.Query) (beegoorm.QuerySeter, error) {
-	qs, err := orm.QuerySetter(ctx, &Artifact{}, query)
+func querySetter(ctx context.Context, query *q.Query, options ...orm.Option) (beegoorm.QuerySeter, error) {
+	qs, err := orm.QuerySetter(ctx, &Artifact{}, query, options...)
 	if err != nil {
 		return nil, err
 	}
diff --git a/src/pkg/task/dao/execution.go b/src/pkg/task/dao/execution.go
index 8cd66e75aa0..6775a7f3ac8 100644
--- a/src/pkg/task/dao/execution.go
+++ b/src/pkg/task/dao/execution.go
@@ -107,7 +107,7 @@ func (e *executionDAO) Count(ctx context.Context, query *q.Query) (int64, error)
 			Keywords: query.Keywords,
 		}
 	}
-	qs, err := e.querySetter(ctx, query)
+	qs, err := e.querySetter(ctx, query, orm.WithSortDisabled(true))
 	if err != nil {
 		return 0, err
 	}
@@ -349,8 +349,8 @@ type jsonbStru struct {
 	value     interface{}
 }
 
-func (e *executionDAO) querySetter(ctx context.Context, query *q.Query) (orm.QuerySeter, error) {
-	qs, err := orm.QuerySetter(ctx, &Execution{}, query)
+func (e *executionDAO) querySetter(ctx context.Context, query *q.Query, options ...orm.Option) (orm.QuerySeter, error) {
+	qs, err := orm.QuerySetter(ctx, &Execution{}, query, options...)
 	if err != nil {
 		return nil, err
 	}
diff --git a/src/pkg/task/dao/task.go b/src/pkg/task/dao/task.go
index b42781c17a9..6bdcec19944 100644
--- a/src/pkg/task/dao/task.go
+++ b/src/pkg/task/dao/task.go
@@ -74,7 +74,7 @@ func (t *taskDAO) Count(ctx context.Context, query *q.Query) (int64, error) {
 			Keywords: query.Keywords,
 		}
 	}
-	qs, err := t.querySetter(ctx, query)
+	qs, err := t.querySetter(ctx, query, orm.WithSortDisabled(true))
 	if err != nil {
 		return 0, err
 	}
@@ -250,8 +250,8 @@ func (t *taskDAO) GetMaxEndTime(ctx context.Context, executionID int64) (time.Ti
 	return endTime, nil
 }
 
-func (t *taskDAO) querySetter(ctx context.Context, query *q.Query) (orm.QuerySeter, error) {
-	qs, err := orm.QuerySetter(ctx, &Task{}, query)
+func (t *taskDAO) querySetter(ctx context.Context, query *q.Query, options ...orm.Option) (orm.QuerySeter, error) {
+	qs, err := orm.QuerySetter(ctx, &Task{}, query, options...)
 	if err != nil {
 		return nil, err
 	}
